openmdao/docs/developer_docs/repo_guide/advanced_operations.rst

```diff
@@ -0,0 +1,234 @@
+without errors on Travis on any/all platforms and Python distributions. If you’re using Sphinx, the docs will build into
+your `<project>/docs/_build/html` directory. The concept is simply to copy the contents of that html directory over to a
+server directory that is serving up html.  For the purposes of OpenMDAO/blue, this happens on `webXXX.webfaction.com:webapps/<doc_serving_app>`
+
+You need to make several changes to your `.travis.yml` file in order to make this happen.
+One of these steps, you will find out, is going to re-write your `.travis.yml` file and take out all the spacing,
+comments, and formatting. So if you like the way your `.travis.yml` file looks, I suggest backing up that file beforehand.
+You can then take the auto-generated lines that we’re about to generate, and stick them back in the pretty-looking `.travis.yml` file.
+
+To get this doc transfer automated, we need to be able to move built docs from Travis to our private server without any required input,
+because there is no human in the loop to type passwords or answer prompts.  This will require some ssh key wizardry.
+The overall concept is to have an encrypted key already on the Travis server, then to decrypt it, and use it to do a
+password-less transfer of the docs from Travis to our private server (in our case, Webfaction).  Here's the outline:
+
+    ::
+
+	1. Generate a dedicated SSH key for this job (it is easier to later isolate and to revoke it, if we ever must).
+	2. Encrypt the private key to make it readable only by Travis CI .
+	3. Copy the public key file into the remote SSH host’s allowed_hosts file.
+	4. Modify the yaml further.
+	5. Commit the modified .travis.yaml and encrypted key file into Git.
+
+
+
+
+Specific Commands
+-----------------
+
+The commands to do the above look something like this:
+
+**1. Generate the Key**
+    Run this command from the top level of your project (in the example case, from the `openmdao/` level):
+
+    ::
+
+        ssh-keygen -t rsa -b 4096 -f ./travis_deploy_rsa
+
+        -t chooses the type of key to generate, in this case, rsa
+        -b  Specifies the number of bits in the key to create.  For RSA keys, the minimum
+            size is 768 bits and the default is 2048 bits. I used 4096 because it was recommended by Travis.
+        -f output into this keyfile, in this case, I want the key created right where I am in the dir structure.
+
+    It will spit out both a private key file and a public key file: `travis_deploy_rsa` and `travis_deploy_rsa.pub`.
+
+**2. Encrypt the Key**
+    Again from the same directory, execute the next command to encrypt the PRIVATE key file that we just generated.
+    (NOTE: this is the aforementioned command that will change your .travis.yml file! ):
+
+    ::
+
+        travis encrypt-file travis_deploy_rsa --add -—debug --org
+
+        —-add adds the decryption command to the .travis.yml file automatically.
+        —-debug shows all the available output of the command.
```

Comment:
some long dashes here instead of two regular dashes

Reply:
Not sure where those come from, but they're fixed.